### DISCUSSION TITLE
Abp Background Jobs

### DISCUSSION THREAD

**Discussion by Osama94** [abpframework/abp#14616] [Q&A]
Abp Background Jobs Increasingly waits between retries for a job. It waits 1 minute for the first retry, 2 minutes for the second retry, 4 minutes for the third retry and so on.

- is it available to enqueue different jobs with different settings like i want job(A) run every 5 min and another job(B)  run every 15 min.. etc ?

**Comment by maliming** [comment]
hi


The default implement is simple. You can consider use Hangfire, RabbitMQ and Quartz.

> Background job system is extensible and you can change the default background job manager with your own implementation or on of the pre-built integrations.

https://docs.abp.io/en/abp/latest/Background-Jobs#integrations

**Comment by mgochmuradov** [comment]
You could replace the built-in implementation with a custom one. Something like:
```cs
[Dependency(ReplaceServices = true)]
public class BackgroundJobWorker : Volo.Abp.BackgroundJobs.BackgroundJobWorker
{
    protected override DateTime? CalculateNextTryTime(BackgroundJobInfo jobInfo, IClock clock)
    {
        var nextTryDate = clock.Now.AddSeconds(60); // will retry every minut
        return nextTryDate;
    }
}
```
